packages/compartment-mapper/test/test-main.js

```diff
@@ -292,3 +294,108 @@ test('no transitive dev dependencies', async t => {
     },
   );
 });
+
+test('makeArchive / parseArchive, but with sha512 corruption', async t => {
+  t.plan(1);
+  await setup();
+
+  const { computeSha512 } = readPowers;
+
+  const corruptSha512 = bytes => {
+    const hash = computeSha512(bytes);
+    return `b4d${hash}`;
+  };
+
+  const archive = await makeArchive(
+    {
+      ...readPowers,
+      computeSha512: corruptSha512,
+    },
+    fixture,
+    {
+      modules,
+      dev: true,
+    },
+  );
+
+  const application = await parseArchive(archive, 'app.agar', {
```

Comment:
What makes the hash bad isn’t so much that it’s inaccurate as that it doesn’t match the one given to the loader in the later steps.

All of the functions that accept a sha512 power _could_ do a trivial check, like `computeSha512(new Uint8Array(0))` and verify that it returns the correct SHA-512 for that or a smattering of known byte sequences. Then it would be possible to fail on `parseArchive`, but, on the other hand, then it wouldn’t be possible to create an invalid archive with `parseArchive` and test the loader’s ability to recognize that it is invalid.